metaci/plan/models.py

```diff
@@ -205,3 +206,49 @@ def get_absolute_url(self):
     @property
     def should_run(self):
         return self.active and self.plan.active
+
+
+class PlanRepositoryTriggerQuerySet(models.QuerySet):
+    def should_run(self):
+        return self.filter(active=True, target_plan_repo__active=True)
+
+
+class PlanRepositoryTrigger(models.Model):
+    source_plan_repo = models.ForeignKey(
+        PlanRepository, on_delete=models.CASCADE, related_name="Sources"
+    )
+    target_plan_repo = models.ForeignKey(
+        PlanRepository, on_delete=models.CASCADE, related_name="Triggers"
+    )
+    branch = models.CharField(max_length=255)
+    active = models.BooleanField(default=True)
+
+    objects = PlanRepositoryTriggerQuerySet.as_manager()
+
+    class Meta:
+        ordering = ["target_plan_repo", "source_plan_repo"]
+        unique_together = ("target_plan_repo", "source_plan_repo")
+        verbose_name_plural = "Plan Repository Triggers"
+        permissions = (
+            ("run_plan", "Run Plan"),
+            ("view_builds", "View Builds"),
+            ("rebuild_builds", "Rebuild Builds"),
+            ("org_login", "Login to Org"),
+        )
+
+    def _get_or_create_branch(self):
+        branch, _ = Branch.objects.get_or_create(
+            repo=self.target_plan_repo.repo, name=self.branch
+        )
+        return branch
+
+    def fire(self, finished_build):
+        build = Build(
+            repo=self.target_plan_repo.repo,
+            plan=self.target_plan_repo.plan,
+            planrepo=self.target_plan_repo,
+            commit=self.branch,
```

Comment:
If the new build doesn't start running immediately, do we want it to use the state of this branch from the time the build was queued or the time it runs? If the former, we should resolve this to an actual commit sha like the plan form does in metaci/plan/forms.py

Reply:
I was operating under the assumption that it was the latter, but resolving to an actual hash would be consistent with our behavior elsewhere, so I'll update.